app/waf/src/controllers/pool.controller.ts

```diff
@@ -161,71 +153,75 @@ export class PoolController {
 
   @get(prefix + '/members', {
     responses: {
-      '200': {
-        description: 'Array of Member model instances',
-        content: {
-          'application/json': {
-            schema: {type: 'array', items: {'x-ts-type': Member}},
-          },
-        },
-      },
+      '200': Schema.collectionResponse(
+        Member,
+        'Successfully retrieve member resources',
+      ),
     },
   })
   async findMembers(
     @param.query.object('filter', getFilterSchemaFor(Member)) filter?: Filter,
-  ): Promise<Member[]> {
-    return await this.memberRepository.find(filter);
+  ): Promise<MemberCollectionResponse> {
+    return new MemberCollectionResponse(
+      await this.memberRepository.find(filter),
+    );
   }
 
+  readonly createMemberDesc = 'Member resource that need to be created';
   @post(prefix + '/pools/{pool_id}/members', {
     responses: {
-      '200': {
-        description: 'Member add to Pool success',
-        content: {'application/json': {schema: {'x-ts-type': Member}}},
-      },
+      '200': Schema.response(Member, 'Successfully create Member resource'),
+      '400': Schema.badRequest('Invalid Member resource'),
+      '422': Schema.unprocessableEntity('Unprocessable Member resource'),
     },
   })
   async createPoolMember(
-    @param.path.string('pool_id') pool_id: string,
-    @requestBody() member: Partial<Member>,
-  ): Promise<Member> {
-    return await this.poolRepository.members(pool_id).create(member);
+    @param(Schema.pathParameter('pool_id', 'Pool resource ID'))
```

Comment:
Suggest to change path param name to poolId, too. It will be exposed by API doc, let us make the naming to be consistent.